lib/metaclient/meta_client.go

```diff
@@ -2002,8 +2002,24 @@ func (c *Client) DeleteMetaNode(id uint64) error {
 	return c.retryUntilExec(proto2.Command_DeleteMetaNodeCommand, proto2.E_DeleteMetaNodeCommand_Command, cmd)
 }
 
+func pingServer(server string) error {
+	pingUrl := server + "/ping"
+	client := http.Client{Timeout: time.Second}
+	req, _ := http.NewRequest("GET", pingUrl, nil)
+	_, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 // CreateSubscription creates a subscription against the given database and retention policy.
 func (c *Client) CreateSubscription(database, rp, name, mode string, destinations []string) error {
+	for _, destination := range destinations {
+		if err := pingServer(destination); err != nil {
+			return errors.New("Fail to ping " + destination)
+		}
+	}
 	return c.retryUntilExec(proto2.Command_CreateSubscriptionCommand, proto2.E_CreateSubscriptionCommand_Command,
```

Comment:
can `retryUntilExec` replace with `retryExec`? We can discuss it.

Reply:
I compared `retryUntilExec` and `retryExec`, `retryUntilExec` is a simple warp of `retryExec`, but `retryUntilExec` wait for local updates (of metadata) to catch up with remote updates. In `showSubscription`, client will only use local cached data, if we use `retryExec`, maybe it sometimes leads to inconsistencies?